applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_axisymmetric_element.cpp

```diff
@@ -38,20 +39,8 @@ void SmallStrainUPwDiffOrderAxisymmetricElement::CalculateBMatrix(Matrix& rB, co
 {
     KRATOS_TRY
 
-    const double radius = GeoElementUtilities::CalculateRadius(Np, this->GetGeometry());
-
-    const SizeType Dim      = this->GetGeometry().WorkingSpaceDimension();
-    const SizeType NumNodes = this->GetGeometry().size();
-
-    for (IndexType i = 0; i < NumNodes; ++i) {
-        const IndexType index = Dim * i;
-
-        rB(INDEX_2D_PLANE_STRAIN_XX, index + INDEX_X) = GradNpT(i, INDEX_X);
-        rB(INDEX_2D_PLANE_STRAIN_YY, index + INDEX_Y) = GradNpT(i, INDEX_Y);
-        rB(INDEX_2D_PLANE_STRAIN_ZZ, index + INDEX_X) = Np[i] / radius;
-        rB(INDEX_2D_PLANE_STRAIN_XY, index + INDEX_X) = GradNpT(i, INDEX_Y);
-        rB(INDEX_2D_PLANE_STRAIN_XY, index + INDEX_Y) = GradNpT(i, INDEX_X);
-    }
+    AxisymmetricStressStatePolicy policy;
+    rB = policy.CalculateBMatrix(GradNpT, Np, this->GetGeometry());
```

Comment:
Right now, it seems this could be improved by making CalculateBMatrix a static function, but the intended design will call a virtual function for CalculateBMatrix on an abstract base-class, which can be easily swapped/mixed&matched with 3d or plane strain versions.

We will have an `std::unique_ptr<StressStatePolicy> mpStressStatePolicy` which can be of type `AxisymmetricStressStatePolicy` or its 3d/plane strain counter parts. Since we need the functions to be virtual in this case, they can't be static.